Src/BootCamp.Chapter/Player.cs

```diff
@@ -29,79 +30,146 @@ public class Player
         /// Player items. There can be multiple of items with same name.
         /// </summary>
         private Inventory _inventory;
+
         /// <summary>
         /// Needed only for the extra task.
         /// </summary>
         private Equipment _equipment;
 
+        public List<Item> Items { get; set; }
+
         public Player()
         {
+            _inventory = new Inventory();
+           _equipment = new Equipment();
         }
 
         /// <summary>
         /// Gets all items from player's inventory
         /// </summary>
-        public Item[] GetItems()
-        {
-            return new Item[0];
-        }
 
         /// <summary>
         /// Adds item to player's inventory
         /// </summary>
         public void AddItem(Item item)
         {
+            _inventory.AddItem(item);
         }
 
         public void Remove(Item item)
         {
-
+            _inventory.RemoveItem(item);
         }
 
         /// <summary>
         /// Gets items with matching name.
         /// </summary>
         /// <param name="name"></param>
-        public Item[] GetItems(string name)
+        public List<Item> GetItems(string name)
         {
-            return new Item[0];
+           return  _inventory.GetItems(name); 
         }
 
         #region Extra challenge: Equipment
+
         // Player has equipment.
         // Various slots of equipment can be equiped and unequiped.
         // When a slot is equiped, it contributes to total defense
         // and total attack.
         // Implement equiping logic and total defense/attack calculation.
-        public void Equip(Headpiece head)
+        private bool CanEquipNewItemt(Item currentEquipedItem, Item newItemToEquip)
```

Comment:
remove the t from the end of the name :)

Reply:
done